### DISCUSSION TITLE
.NET Core 6

### DISCUSSION THREAD

**Discussion by Whoome** [neolithos/neolua#141] [General]
I started to try and compile this for .NET core 6, and am running into the problem that LambdaExpression.Compile to method has been removed from the framework (for unclear reasons).  It looks like compiling to a delegate would still be possible, but the handling of constants in the expression would somehow need to change due to no longer having a type.   

Unfortunately, I'm not quite clear on how that might work - any thoughts?

**Comment by neolithos** [comment]
[MSDN ](https://docs.microsoft.com/en-us/dotnet/api/system.linq.expressions.lambdaexpression.compile?view=net-6.0#System_Linq_Expressions_LambdaExpression_Compile) say it is still there?

**Comment by Whoome** [comment]
Nah... look closer MSDN doesn't even list CompileToMethod.   Sorry, I put spaces in the name yesterday.  My fault.   That's the thing that has gone missing.   Compile is still there, but that doesn't generate IL into a type object, which seems to be where constants are stored?

**Comment by Whoome** [comment]
BTW.... the super weird bit is that .NET-5 compiles with that method and then the most recent .NET 6 doesn't.   So.... I'm a bit confused and can't find any documentation about why it has vanished.

**Comment by Whoome** [comment]
Ah... it looks like this was compiled out (since it is the debug trace.  I can just fixup the conditional.

**Reply by neolithos** [reply]
Thanks for looking into this, I am really busy, currently.
